HousingFinanceInterimApi/V1/Gateways/LogParserGateway.cs

```diff
@@ -0,0 +1,121 @@
+using Amazon.CloudWatchLogs.Model;
+using HousingFinanceInterimApi.V1.Gateway.Interfaces;
+using HousingFinanceInterimApi.V1.Infrastructure;
+using HousingFinanceInterimApi.V1.Handlers;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+
+namespace HousingFinanceInterimApi.V1.Gateways
+{
+    public class LogParserGateway : ILogParserGateway
+    {
+        private readonly IDatabaseContext _context;
+
+        public LogParserGateway(IDatabaseContext context)
+        {
+            _context = context;
+        }
+
+        public async Task UpdateDatabaseWithResults(string logGroupName, List<List<ResultField>> queryResults)
+        {
+            // Validate input parameters
+            if (string.IsNullOrWhiteSpace(logGroupName))
+            {
+                throw new ArgumentNullException(nameof(logGroupName), "Log group name cannot be null or empty.");
+            }
+
+            // Handle null or empty queryResults
+            if (queryResults is null || !queryResults.Any())
+            {
+                await LogResultAsync(logGroupName, DateTime.UtcNow, true).ConfigureAwait(false);
+                return;
+            }
```

Comment:
Just a note:
Mmm... :thinking: ... if the query for failures ends up missing out on some failure-related keywords, this will end up giving false positives. So with this implementation, the query has to be as exhaustive as it can be.

Reply:
I am hoping that we can rely on CW Insights filtering to be able to return all log group messages containing the keyword "error". I have noticed that this word is used for all lambda function failures...
Fingers crossed!